implementations/rust/kex/src/xx.rs

```diff
@@ -106,31 +111,57 @@ impl KeyExchange for SymmetricState {
         &mut self,
         secret_handle: SecretKeyContext,
         public_key: PublicKey,
-    ) -> Result<Vec<u8>, VaultFailError> {
+    ) -> Result<(SecretKeyContext, SecretKeyContext), VaultFailError> {
+        let ck = self
+            .state
+            .ck
+            .ok_or(VaultFailError::from(VaultFailErrorKind::Ecdh))?;
+
         let mut vault = self.vault.lock().unwrap();
-        vault.ec_diffie_hellman_hkdf_sha256(
+
+        let attributes_ck = SecretKeyAttributes {
+            xtype: SecretKeyType::Buffer(32),
```

Comment:
```suggestion
            xtype: SecretKeyType::Aes256,
```

Reply:
@mikelodder7 for atecc608a I ended up creating new secret key type called "ChainKey", probably I'll do the same thing for rust later. Before that I don't understand why would chain key be of AES256 type